cli/src/cli_util.rs

```diff
@@ -634,6 +641,7 @@ impl ReadonlyUserRepo {
         }
     }
 
+    #[cfg(feature = "git")]
```

Comment:
nit: does it help if we had a dummy `git_utils` or something that stubs out various git functions and types?

It might be also good to extract git-related functions to separate `impl` block, and feature-gate it.

Reply:
I tried this approach and was able to reduce the amount of `#[cfg]` junk a bit. I think we need to just keep cleaning things up. It seems like we're probably even pretty close to being able to put the `git`/`git_util` code into a separate crate, and then feed it into `jj-cli`? Maybe I can take a stab at it.

Reply:
Looks better, thanks.

> It seems like we're probably even pretty close to being able to put the `git`/`git_util` code into a separate crate, and then feed it into `jj-cli`?

I think many feature-gated jj-lib modules can be extracted to jj-lib-git or something (but we might have to extract core/utility types first to re-export `jj-lib-git::*` from jj-lib and/or run jj-lib tests.) I have no idea about jj-cli part.

btw, dummy type could be an empty enum to ensure that it'll never be constructed.

Reply:
I tried a little last night and was able to make a start on a `jj-git` crate that moves some of the code in there and some tests there. We'll probably have to do a series of moves like you mention, for a types-based crate. I'll submit another change later.